modules/hivemq-edge-module-etherip/src/main/java/com/hivemq/edge/adapters/etherip/model/EtherIpValueFactory.java

```diff
@@ -0,0 +1,53 @@
+package com.hivemq.edge.adapters.etherip.model;
+
+import com.hivemq.edge.adapters.etherip.model.dataypes.EtherIpBool;
+import com.hivemq.edge.adapters.etherip.model.dataypes.EtherIpDouble;
+import com.hivemq.edge.adapters.etherip.model.dataypes.EtherIpInt;
+import com.hivemq.edge.adapters.etherip.model.dataypes.EtherIpLong;
+import com.hivemq.edge.adapters.etherip.model.dataypes.EtherIpString;
+import etherip.types.CIPData;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Optional;
+
+public class EtherIpValueFactory {
+    private static final Logger log = LoggerFactory.getLogger(EtherIpValue.class);
+
+    public static Optional<EtherIpValue> fromTagAddressAndCipData(@NotNull String tagAddress, @NotNull CIPData cipData) {
+        CIPData.Type dataType = cipData.getType();
+
+        try {
+            if(cipData.isNumeric()) {
+                if (cipData.getElementCount() > 1) {
+                    log.warn("More than one element returned, only the first one will be used");
+                }
+                Number number = cipData.getNumber(0);
+                log.debug("Got value {} for type {} for tag address {}", number, dataType, tagAddress);
+                switch (dataType) {
+                    case BOOL:
+                        return Optional.of(new EtherIpBool(tagAddress, number));
```

Comment:
Great way of doing it. Making reading every easy.

Reply:
Thanks :)